### PR TITLE
feat(actions): requery the element when it was detached during the action

### LINKED ISSUE #1780: [Feature] page.click(selector) should skip intermediate elements
When element with given selector appears and disappears w/o reaching the interactable state, skip it and wait for a new element.

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #1780.

### CODE REVIEW

File: src/frames.ts [hunk 1]
```diff
@@ -693,16 +693,34 @@ export class Frame {
...
+      try {
+        const { world, task } = selectors._waitForSelectorTask(selector, 'attached', deadline);
+        const handle = await this._scheduleRerunnableTask(task, world, deadline, `selector "${selector}"`);
+        const element = handle.asElement() as dom.ElementHandle<Element>;
+        const result = await action(element, deadline);
+        element.dispose();
+        return result;
+      } catch (e) {
+        if (options.force || !(e instanceof NotConnectedError))
+          throw e;
```
Comment (pavelfeldman) [inline src/frames.ts:710]: it'd be nice to dispose the handle to the detached element - we are likely the only client retaining it.

File: src/frames.ts [hunk 2]
```diff
@@ -717,10 +735,9 @@ export class Frame {
     handle.dispose();
   }
 
-  async hover(selector: string, options?: dom.PointerActionOptions & types.PointerActionWaitOptions) {
-    const { handle, deadline } = await this._waitForSelectorInUtilityContext(selector, options);
```
Comment (pavelfeldman) [inline src/frames.ts:721]: Do we still have users of _waitForSelectorInUtilityContext?

**Review by pavelfeldman — APPROVED** [review]

File: src/frames.ts (hunk 2 above)
Comment (dgozman) [reply src/frames.ts:721]: Not anymore.

File: src/frames.ts (hunk 1 above)
Comment (dgozman) [reply src/frames.ts:710]: Done.

**Pull request merged**